common/src/main/java/org/apache/gravitino/dto/requests/PolicyUpdateRequest.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.dto.requests;
+
+import static org.apache.gravitino.dto.util.DTOConverters.fromDTO;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.google.common.base.Preconditions;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.gravitino.dto.policy.PolicyContentDTO;
+import org.apache.gravitino.policy.PolicyChange;
+import org.apache.gravitino.rest.RESTRequest;
+
+/** Represents a request to update a policy. */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY)
+@JsonSubTypes({
+  @JsonSubTypes.Type(value = PolicyUpdateRequest.RenamePolicyRequest.class, name = "rename"),
+  @JsonSubTypes.Type(
+      value = PolicyUpdateRequest.UpdatePolicyCommentRequest.class,
+      name = "updateComment"),
+  @JsonSubTypes.Type(
+      value = PolicyUpdateRequest.UpdatePolicyContentRequest.class,
+      name = "updateContent")
+})
+public interface PolicyUpdateRequest extends RESTRequest {
+
+  /**
+   * Returns the policy change.
+   *
+   * @return the policy change.
+   */
+  PolicyChange policyChange();
+
+  /** The policy update request for renaming a policy. */
+  @EqualsAndHashCode
+  @Getter
+  @ToString
+  class RenamePolicyRequest implements PolicyUpdateRequest {
+
+    @JsonProperty("newName")
+    private final String newName;
+
+    /**
+     * Creates a new RenamePolicyRequest.
+     *
+     * @param newName The new name of the policy.
+     */
+    public RenamePolicyRequest(String newName) {
+      this.newName = newName;
+    }
+
+    /** This is the constructor that is used by Jackson deserializer */
+    public RenamePolicyRequest() {
```

Comment:
Make all the jackson default constructor private, also using `this(xxx, xxx...)` to replace the code.